apps/web/pages/[user].tsx

```diff
@@ -195,9 +193,38 @@ const getEventTypesWithHiddenFromDB = async (userId: number) => {
   }));
 };
 
-export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+export type UserPageProps = {
+  trpcState: DehydratedState;
+  profile: {
+    name: string;
+    image: string;
+    theme: string | null;
+    brandColor: string;
+    darkBrandColor: string;
+  };
+  users: Pick<User, "away" | "name" | "username" | "bio" | "verified">[];
+  themeBasis: string | null;
+  markdownStrippedBio: string;
+  safeBio: string;
+  eventTypes: ({
+    descriptionAsSafeHTML: string;
+    metadata: z.infer<typeof EventTypeMetaDataSchema>;
+  } & Pick<
+    EventType,
+    | "id"
+    | "title"
+    | "slug"
+    | "length"
+    | "hidden"
+    | "requiresConfirmation"
+    | "price"
+    | "currency"
+    | "recurringEvent"
+  >)[];
+} & EmbedProps;
+
+export const getServerSideProps: GetServerSideProps<UserPageProps> = async (context) => {
```

Comment:
Explicitly declaring the return props gives both a degree of safety and also speeds up typescript